api/sys_raft.go

```diff
@@ -0,0 +1,98 @@
+package api
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+)
+
+type RaftJoinResponse struct {
+	Joined bool `json:"joined"`
+}
+
+type RaftJoinRequest struct {
+	LeaderAddr string `json:"leader_api_addr"`
+	CACert     string `json:"ca_cert":`
+	Retry      bool   `json:"retry"`
+}
+
+func (c *Sys) RaftJoin(opts *RaftJoinRequest) (*RaftJoinResponse, error) {
+	r := c.c.NewRequest("POST", "/v1/sys/storage/raft/join")
+
+	if err := r.SetJSONBody(opts); err != nil {
+		return nil, err
+	}
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+	resp, err := c.c.RawRequestWithContext(ctx, r)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	var result RaftJoinResponse
+	err = resp.DecodeJSON(&result)
+	return &result, err
+}
```

Comment:
I don't think we need a specific Join or Config function here, they can just reuse the `Logical().Write()` call, right?

Reply:
I have removed the function `RaftConfig` function but have retained the `RaftJoin` function. The reason is that the join API is not a logical one. It is defined directly on the mux and is implemented at the HTTP layer, which means it won't return a `*logical.Response` expected by the `client.Logical()`.